cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java

```diff
@@ -771,9 +812,17 @@ private Principal getCurrentUserID()
 
         String roleString = role.getValue();
         URL searchURL = lookupServiceURL(Standards.GMS_SEARCH_01);
-        URL getMembershipsURL = new URL(searchURL.toExternalForm()
-                                        + "?ROLE="
-                                        + NetUtil.encode(roleString));
+        StringBuilder sb = new StringBuilder();
```

Comment:
The above Standards.GMS_SEARCH_01 is not the correct GMS-1.0 standard ID. The code below uses the old prototype which is obsolete. How does this actually work?

Should be GMS_SEARCH_10; read the spec about output format. You probably need to rewrite this method, in which case the input List<GroupURI> and "ignoring" can happen right in the loop that appends `group` params (line ~820).

Reply:
My test VM has both Standards: GMS_SEARCH_01 & GMS_SEARCH_10 which is how the int-tests run okay. I'll remove the old Standard.
Yes better that the the group serviceID checking happens in the 'low level' method so the same check doesn't have to happen in multiple places.
The output of that method looks correct, a list of group names the caller belongs to. But it, and several other methods, are using the deprecated HttpDownload instead of HttpGet. Missed that before.

Reply:
Left the `ignore` checking in the public `getMemberships` method as an optimization. If there are no groups left after the ignore check, an empty list can be returned, rather than do the LDAP query, which will just return an empty list.